Formula/gz-ionic.rb

```diff
@@ -0,0 +1,54 @@
+class GzIonic < Formula
+  include Language::Python::Virtualenv
+
+  desc "Collection of gazebo simulation software"
+  homepage "https://github.com/gazebosim/gz-ionic"
+  url "https://github.com/gazebosim/gz-ionic.git", branch: "main"
+  version "0.999.999-0-20231016"
+  license "Apache-2.0"
+
+  head "https://github.com/gazebosim/gz-ionic.git", branch: "main"
+
+  depends_on "cmake" => :build
+  depends_on "python@3.11" => [:build, :test]
+
+  depends_on "gz-cmake4"
+  depends_on "gz-common6"
+  depends_on "gz-fuel-tools10"
+  depends_on "gz-gui9"
+  depends_on "gz-launch8"
+  depends_on "gz-math8"
+  depends_on "gz-msgs11"
+  depends_on "gz-physics8"
+  depends_on "gz-plugin3"
+  depends_on "gz-rendering9"
+  depends_on "gz-sensors9"
+  depends_on "gz-sim9"
+  depends_on "gz-tools2"
+  depends_on "gz-transport14"
+  depends_on "gz-utils3"
+  depends_on "pkg-config"
+  depends_on "sdformat15"
+
+  resource "PyYAML" do
+    url "https://files.pythonhosted.org/packages/64/c2/b80047c7ac2478f9501676c988a5411ed5572f35d1beff9cae07d321512c/PyYAML-5.3.1.tar.gz"
+    sha256 "b8eac752c5e14d3eca0e6dd9199cd627518cb5ec06add0de9d32baeee6fe645d"
+  end
+
+  def install
+    mkdir "build" do
+      system "cmake", "..", *std_cmake_args
+      system "make", "install"
+    end
+
+    venv = virtualenv_create(libexec, Formula["python@3.11"].opt_libexec/"bin/python")
+    %w[PyYAML vcstool].each do |pkg|
+      venv.pip_install pkg
+    end
+  end
+
+  test do
+    yaml_file = share/"gz/gz-ionic/gazebodistro/collection-ionic.yaml"
+    system libexec/"bin/vcs", "validate", "--input", yaml_file
```

Comment:
we should change this test if we are installing release notes instead of a yaml file